### DISCUSSION TITLE
Mensaje a desarrolladores

### DISCUSSION THREAD

**Discussion by Raimundo-ISPC** [ispc-programador2022/a2g3#26] [General]
# Mensaje a desarrolladores

Estimados,

Veo que muchos han creado funciones sin la correcta recepción de parámetros.

Los parámetros, son valores que la función recibe desde "afuera" de sí misma. 
No es la función la que debe crear dichos valores (salvo que el enunciado de su creación así lo solicite).

Si bien un `input()` permite que uno "ingrese" valores a la función, lo cierto es que un parámetro, es un valor ya existente antes que la función sea llamada. 

Debemos pensar la función como una máquina que funcionará con datos que recibirá externamente cuando sea llamada para tal fin.

El modo de declarar una función con parámetros es:

`def nombreDeMiFuncion(parametro1,parametro2,parametroN):`

Y una forma interesante de pensar si nuestra función está correctamente escrita es:

¿funcionará correctamente cuando sea llamada desde otra función para hacer la tarea que se espera que realice?

Porque muchas funciones requieren que otras funciones estén bien escritas para funcionar correctamente.

Sino, el programa colapsará al invocar dicha función.

Es decir... Puede ser que de manera independiente la función "funcione" y devuelva un resultado "correcto" si le pedimos al usuario que ingrese valores desde afuera cuando se le solicita, pero imaginen una calculadora que cuando intentamos que realice una suma por la cual ya ingresamos los números que se espera que sume, no nos arroja el resultado sino que nos dice: "Ingresa tu numero" o algo así. No! Nosotros ya le pusimos los números que debía sumar antes de pulsar el símbolo de igual y queremos el resultado, no una nueva pregunta sobre qué numero deseo sumar. 

Es un ejemplo un poco tosco, pero espero que sirva para orientar a quienes no tienen experiencia en programación. 

Resulta un poco complejo (a veces) trabajar con cosas un tanto abstractas, (les juro que a mi también me cuesta mucho), pero debemos pensar que cuando otro desarrollador deba llamar a nuestra función para obtener cierto resultado, ¿lo podrá hacer sin interrupciones? 

Espero que se comprenda el concepto. Gracias!

**Comment by Raimundo-ISPC** [comment]
En muchas funciones se requiere del uso de funciones que ya han sido creadas por otro desarrollador en el proyecto.

Para invocar una función, por ejemplo: suma(), se debe hacer desde el archivo suma.py y se hace del siguiente modo:
`from suma import suma`

Una vez hecho esto, podremos hacer uso de la función suma() que se encontraba en el archivo suma.py (la extensión del archivo ".py" no se escribe.

**Comment by Raimundo-ISPC** [comment]
En el ejemplo de la función suma() debiera ser algo así:

`def funcion suma(param1, param2):
    return param1 + param2`

Con eso sería suficiente. 

Al ser llamada para la función de sumar los dos números que se deseen, simplemente nos devolverá con el: `return` el resultado de dicha suma!

El nombre del parámetro en sí es indistinto. 

La función podría funcionar perfectamente así:

```
def suma(sandwichDeBondiola, grandeDeJamonYMorrones);
    return sandwichDeBondiola + grandeDeJamonYMorrones
```
Y funcionará perfectamente!

El uso de "param1" y "param2" es sólo a los efectos de hacer más claro de qué estamos hablando y lo ideal es poner nombres cortos y descriptivos. 

Pero, en este caso, la función arrojará el resultado correctamente.

**Comment by Carolina-Bolatti** [comment]
Excelente @Raimundo-ISPC lo tendré en cuenta, espero poder aplicarlo bien y que el resto también lo haga! Muy bien explicado!

**Comment by Estannoy** [comment]
Una consulta: ¿Es necesario que todos los parámetros tengan el mismo nombre en todos los archivos del proyecto?
